packages/pretty-format/src/plugins/dom_element.js

```diff
@@ -37,17 +37,23 @@ type Comment = {
   data: string,
   nodeType: 8,
 };
+type Fragment = {
+  children: Array<Element>,
```

Comment:
Not sure how important typing this is? It's not an `Array`, it's an `HTMLCollection`, but the other typing here is pretty bad anyways

Reply:
using the types from flow makes the other stuff fail here since we can't use `instanceof` checks and not explode outside of jsdom 🤷‍♂️

Reply:
FWIW, flow passes with this diff

```diff
diff --git i/packages/pretty-format/src/plugins/dom_element.js w/packages/pretty-format/src/plugins/dom_element.js
index 26f0801f6..59957aa54 100644
--- i/packages/pretty-format/src/plugins/dom_element.js
+++ w/packages/pretty-format/src/plugins/dom_element.js
@@ -18,30 +18,6 @@ import {
   printText,
 } from './lib/markup';
 
-type Attribute = {
-  name: string,
-  value: string,
-};
-
-type Element = {
-  attributes: Array<Attribute>,
-  childNodes: Array<Element | Text | Comment>,
-  nodeType: 1,
-  tagName: string,
-};
-type Text = {
-  data: string,
-  nodeType: 3,
-};
-type Comment = {
-  data: string,
-  nodeType: 8,
-};
-type DocumentFragment = {
-  children: Array<Element>,
-  nodeType: 11,
-};
-
 const ELEMENT_NODE = 1;
 const TEXT_NODE = 3;
 const COMMENT_NODE = 8;
@@ -76,17 +52,17 @@ export const serialize = (
   refs: Refs,
   printer: Printer,
 ): string => {
-  if (node.nodeType === TEXT_NODE) {
+  if (node instanceof Text) {
     return printText(node.data, config);
   }
 
-  if (node.nodeType === COMMENT_NODE) {
+  if (node instanceof Comment) {
     return printComment(node.data, config);
   }
 
   let type;
 
-  if (node.nodeType === FRAGMENT_NODE) {
+  if (node instanceof DocumentFragment) {
     type = `Fragment`;
   } else {
     type = node.tagName.toLowerCase();
```

Reply:
nice, maybe it's worth to use those?

Reply:
It only works with `instanceof` checks, which means it doesn't work in node, just jsdom